widget.zone-posts.php

```diff
@@ -57,17 +57,15 @@ function widget( $args, $instance ) {
 
 		<?php if ( ! empty( $zone->description ) && $show_description ) : ?>
 			<p class="description"><?php echo esc_html( $zone->description ); ?></p>
-		<?php endif; ?>
-
-		<ul>
-			<?php foreach ( $posts as $post ) : ?>
-				<li>
-					<a href="<?php echo esc_url( get_permalink( $post->ID ) ); ?>">
-						<?php echo esc_html( get_the_title( $post->ID ) ); ?>
-					</a>
-				</li>
-			<?php endforeach; ?>
-		</ul>
+		<?php endif;
+
+        if ( has_filter( 'widget_zone_posts_renderer' ) ) {
+            apply_filters('widget_zone_posts_renderer', $posts);
+        }
```

Comment:
Cool idea. A better pattern I've found is to have `false` value passed to a filter and if unchanged, do the default render; something like:

```
$html = apply_filters( 'widget_zone_posts_override_html', false, $posts );
if ( $html !== false ) {
   echo $html;
} else {
   $this->widget_posts( $posts );
}
```

This way you're not echoing inside a filter which can be dangerous.